core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/encrypter/EncryptionController.java

```diff
@@ -0,0 +1,35 @@
+package io.metadew.iesi.server.rest.encrypter;
+
+import io.metadew.iesi.server.rest.encrypter.dto.EncryptDto;
+import io.metadew.iesi.server.rest.encrypter.dto.EncryptPostDto;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+/**
+ * The encryption controller is used to provide encrypted text.
+ * @author Suyash.d.jain
+ */
+
+@RestController
+@Tag(name = "encrypt", description = "controller to encrypt text")
+@RequestMapping("/encrypt")
```

Comment:
root endpoint should be a subject and in plural like "encryptions". Then, the post mapping should be the action such as "/encrypt". This way, we encrypt values by calling : /encryptions/encrypt. In the future we could have multiple services, so that controller should be generic.

We do the same with the login. Instead of creating a controller /login we do /users/login

Reply:
Login controller is different and Encrypt is different. and For Encryption we are not  going to create and update operation. So This will remain as it is. Have a look on Script and Connection controller, we don't write scripts/create, just /scripts and /connection to post new entity

Reply:
Ok I agree with you first statement. But we could have in the future a method to get a text from an encrypted value. So the root endpoint should be /encryptions and not /encrypt. 

This way for : 
- Encrypting data we do : POST /encryptions/encrypt
- Decrypting data we do : POST /encryptions/decrypt

If we only create a controller named "encrypt", we assume that the controller ONLY encrypt data and nothing other.